### PR TITLE
Eric

### PR DESCRIPTION
**Pull request by QuanlingZhao**
Added European Language HDC

### CODE REVIEW

File: HD/Language_HDC/Language_HDC.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+* [Code](https://github.com/abbas-rahimi/HDC-Language-Recognition)
+
+
+### Motivation and Problem Formulation
+
+* **Motivation**
+  * HDC is a computation paradigm that highly parallel and requires simple arithmetic operations whick means it can be greatly accelerated with specialized hardware design. This early papers on HDC proposed an HDC classifier hardware design that aming for a energy efficient design.
+
+* **Related works**
+    * Hyperdimensional computing: Brain inspired computing model using vectors of very hihg dimensionality. It has features such as robust in noise and extremly low latency computation.
```
Comment (Orienfish) [inline HD/Language_HDC/Language_HDC.md]: typo: hihg -> high

File: HD/Language_HDC/Language_HDC.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    * Vector Symbolic Architecture: A well defined set of operation on large vector, in HDC bundling, binding and perutation are used to combine, bind information.
+    * Memory Centric Design: Hardware design technique where computation components are build in and around memory to bypass the expensive and high latency data transfer limitation of traditional computer architechture.
+
+* **Challenges**
+    * A hardware design cover all componenets required by HDC classification task.
```
Comment (Orienfish) [inline HD/Language_HDC/Language_HDC.md]: typo: componenets -> components

File: HD/Language_HDC/Language_HDC.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    * Properly define operations on hypervector (Bundling, Binding, Permutation)
+    * Efficient implementation on those operations with fast computation and memory access.
+    * Gain efficiency with minimum loss in accuracy.
+
+### Method
+
+* **HD Operation and Encoding Process**
+  * Bundling: Compoenet-wise addition.
```
Comment (Orienfish) [inline HD/Language_HDC/Language_HDC.md]: typo: compoenet -> component
Same for the next two lines

File: HD/Language_HDC/Language_HDC.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+  * Permutation: Compoenet-wise rotate.
+  * For efficiency and computation simplicity, all hypervector are in binary form. \
+    ![Encoding](./Encoding.PNG)
+  * To preserve sequential info, binding and permutation are applied.
+  * Classification are done with cosine similarity check. \
+    ![Similarity](./Similarity.PNG)
+
+![Overall_Design](./System_Design.PNG)
+* **Hardware Design**
+  * HDC classification mimimally requires encoder and simularity checker to carry out classification task.
```
Comment (Orienfish) [inline HD/Language_HDC/Language_HDC.md]: typo: mimimally? simularity?

File: HD/Language_HDC/Language_HDC.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    * In Language Classification:
+      * Trigram are input, querry corresponding letter hypervector and apply binding and permutation.
```
Comment (Orienfish) [inline HD/Language_HDC/Language_HDC.md]: typo: querry?

Comment (Orienfish) [reply HD/Language_HDC/Language_HDC.md]: Also, what is trigram?

File: HD/Language_HDC/Language_HDC.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+  * Result shows proposed hardware design can greatly acclerate HDC computation with small accuracy trade-off.
+  * Memory usage increases exponetially in baseline method, whereas HDC requires constant memory.
+  * Robust against faulty memory cell due to high dimensionality and holographic representation of HDC.
+
+
+### Pros and Cons
+
+* Pros:
+  * Generic design, can generalize to other applications.
+  * Mnemonic centric, bypass BUS data transfer limitation.
```
Comment (Orienfish) [inline HD/Language_HDC/Language_HDC.md]: typo: mnemonic?

File: HD/Language_HDC/Language_HDC.md [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+* **Evaluation Setup**
+  * Test on Eurpoean Language dataset
+    * Texts Sample of 21 different language
+  * Baseline: KNN clustering with N-gram Histogram.
+
+
+
+* **Evaluation Result**
+  ![Result](./Result.PNG)
+  * Result shows proposed hardware design can greatly acclerate HDC computation with small accuracy trade-off.
```
Comment (Orienfish) [inline HD/Language_HDC/Language_HDC.md]: typo: acclerate

File: HD/Language_HDC/Language_HDC.md [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+### Pros and Cons
+
+* Pros:
+  * Generic design, can generalize to other applications.
+  * Mnemonic centric, bypass BUS data transfer limitation.
+  * Take faulty hardware into consideration.
+
+
+* Cons:
+  * Simularity check seems computationally complex, why not hamming distance.
```
Comment (Orienfish) [inline HD/Language_HDC/Language_HDC.md:72]: typo: simularity

File: HD/Language_HDC/Language_HDC.md [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+  * Arbitrary Spares/Dense hypervector decison with no explaination.
```
Comment (Orienfish) [inline HD/Language_HDC/Language_HDC.md]: typo: spares -> sparse, decison -> decision, explaination -> explanation

**Review by Orienfish — CHANGES_REQUESTED** [review]

File: HD/Language_HDC/Language_HDC.md (hunk 5 above)
Comment (QuanlingZhao) [reply HD/Language_HDC/Language_HDC.md]: Add trigram explanation (N-gram length of 3)

File: HD/Language_HDC/Language_HDC.md (hunk 9 above)
Comment (QuanlingZhao) [reply HD/Language_HDC/Language_HDC.md]: Sorry for all the typo....

**Pull request merged**